src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -6,18 +6,46 @@
  */
 public class MatrixOperation {
     public int[] getDiagonal(int[][] matrix) {
-        return null;
+        int[] diagonal = new int[matrix.length];
+        int counter1 = 0;
+        for (int i = 0; i < matrix.length; i++) {
+            diagonal[i] = matrix[i][counter1];
+            counter1++;
+        }
+        return diagonal;
     }
 
     public int[] getCounterDiagonal(int[][] matrix) {
-        return null;
+        int[] counterDiagonal = new int[matrix.length];
+        int counter2 = 0;
```

Comment:
the name of this variable isn't understandable, let's change it
and let's move the declaration of this variable in for loop too
Example:
https://www.geeksforgeeks.org/loop-java-important-points/
https://stackoverflow.com/a/14752913